sim/pokemon.ts

```diff
@@ -1450,6 +1451,24 @@ export class Pokemon {
 		return true;
 	}
 
+	updateMaxHp() {
+		const newMaxHP = Math.floor(Math.floor(
+			2 * this.species.baseStats['hp'] + this.set.ivs['hp'] + Math.floor(this.set.evs['hp'] / 4) + 100
+		) * this.level / 100 + 10);
+		if (this.baseMaxhp === newMaxHP) return;
+		this.baseMaxhp = newMaxHP;
+		this.hp = newMaxHP - (this.maxhp - this.hp);
+		if (this.hp < 0) {
+			if (!this.fainted) {
+				// should never happen
+				throw new Error(`${this.name} is not fainted and suffering a max HP decrease from ${this.baseMaxhp} to ${newMaxHP}`);
```

Comment:
I would be tempted to clamp the `hp` to `1` in this case in case someone devises an OM with some weird forme-changing mechanic that would trigger this.